jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JGivenTestsPublisher.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * The MIT License Copyright (c) 2016, CloudBees, Inc. Permission is hereby
+ * granted, free of charge, to any person obtaining a copy of this software and
+ * associated documentation files (the "Software"), to deal in the Software
+ * without restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions: The above copyright notice and this
+ * permission notice shall be included in all copies or substantial portions of
+ * the Software. THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO
+ * EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES
+ * OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.maven.publishers;
+
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.jgiven.JgivenReportGenerator;
+import org.jenkinsci.plugins.jgiven.JgivenReportGenerator.ReportConfig;
+import org.jenkinsci.plugins.pipeline.maven.MavenPublisher;
+import org.jenkinsci.plugins.workflow.steps.StepContext;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.w3c.dom.Element;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.annotation.Nonnull;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Run;
+import hudson.model.StreamBuildListener;
+import hudson.model.TaskListener;
+
+/**
+ * @author <a href="mailto:cleclerc@cloudbees.com">Cyrille Le Clerc</a>
+ */
+public class JGivenTestsPublisher extends MavenPublisher {
+    private static final Logger LOGGER = Logger.getLogger(JGivenTestsPublisher.class.getName());
+
+    private static final long serialVersionUID = 1L;
+
+    @DataBoundConstructor
+    public JGivenTestsPublisher() {
+
+    }
+
+    @Override
+    public void process(@Nonnull final StepContext context, @Nonnull final Element mavenSpyLogsElt)
+            throws IOException, InterruptedException {
+
+        TaskListener listener = context.get(TaskListener.class);
+        if (listener == null) {
+            LOGGER.warning("TaskListener is NULL, default to stderr");
+            listener = new StreamBuildListener((OutputStream) System.err);
+        }
+
```

Comment:
Can we add a test to check the existence of JGiven reports before writing trying to load the Jenkins JGiven plugin and writing the message "[withMaven] Jenkins JGiven Plugin not found, do not archive jgiven reports."